### DISCUSSION TITLE
X-CIGALE fitting : solving the FIR/sub-mm problems

### DISCUSSION THREAD

**Discussion by Couleyr** [mboquien/cigale#7] [Q&A] [answered]
As shown, we attempted to fit the SED of type-2 AGN with X-CIGALE. 

First we fit the SED without X-ray, and then we fit X-ray with this SED optimal parameter configuration (because this source is highly obscured and we can't make accurate correction to the X-ray data under the existing observations).

It shows that the FIR/sub-mm does not fit well (the first figure). We want to know how setting parameters can solve this problem.

[1266_best_model.pdf](https://github.com/mboquien/cigale/files/8449778/1266_best_model.pdf)

[1266_best_model.pdf](https://github.com/mboquien/cigale/files/8449792/1266_best_model.pdf)

**Comment by Guang91** [comment]
The 2nd figure looks better. Except that the 2nd figure doesn't use X-ray, is there any other difference between the two figures

**Reply by Couleyr** [reply]
In the second figure we get the optimal parameter settings for modules other than the X-ray module, and then we use these parameters for the new fitting process with the addition of the X-ray module (shown in Figure 1).
In Figure 1, however, we don't know what caused the FIR/sub-mm band to not fit.

**Reply by Guang91** [reply]
That's weird. Could you share the .ini and data file?

**Reply by Couleyr** [reply]
Here are my two files
[obs.txt](https://github.com/mboquien/cigale/files/8456033/obs.txt)

[pcigale.txt](https://github.com/mboquien/cigale/files/8456034/pcigale.txt)

**Reply by Guang91** [reply]
The files look okay to me. Just a small comment: you may want to set "redshift_decimals" to a negative value, which will use precise redshift.

**Reply by Couleyr** [reply]
You mean to change the redshift (0.0072) in file obs.txt to -0.0072 ?

**Reply by Guang91** [reply]
No, the "redshift_decimals" parameter in pcigale.txt.

**Reply by Couleyr** [reply]
the red-shifting module setting?

**Reply by Guang91** [reply]
No, under [analysis_params]. 
Your current value is "redshift_decimals = 2". I suggest setting to "redshift_decimals = -1" (or any other negative value).

**Reply by Couleyr** [reply]
Okay, thank you very much Mr. Yang.

**Comment by Guang91** [comment, marked as answer]
The cause might be this: your X-ray flux is too low, and that single data point affects the normalization of the entire model. Imagine you move up the normalization of the model in figure 1, that would make the deviation in X-ray band worse and incur a huge increase in chi2. 

So, I guess the key issue is X-ray data is far too low (and it messed up the entire fit). What's your purpose? If you just need to visually show X-ray is heavily obscured, you can simply include X-ray module in your fits of Figure 2 (but without supplying the X-ray data). Then you will still have the full model (including X-ray) in the output, and just overplot the actual X-ray flux manually.

**Reply by Couleyr** [reply]
Our goal is to obtain the SFR and AGN luminosity of the corresponding source, but X-ray data are not used in Figure 2. 
Therefore, we wanted to perform the fitting again with the help of the existing X-ray data.
However, this source is highly obscured (N_H ~ 10^24) and we cannot give correct corrections. 
The X-ray flux we use here is derived from the AGN luminosity (assuming L_AGN/L_X = 28, and N_H = 6 * 10^24).

**Reply by Guang91** [reply]
To use the X-ray data, you will need to correct obscuration. Since it's in Compton-thick regime, such a correction is not trivial. You need modeling the X-ray spectrum with XSPEC (or something else) and derive the obscuration correction. Otherwise, the X-ray flux cannot be used.

**Reply by Guang91** [reply]
One thing I'm still puzzled is that you're saying "The X-ray flux we use here is derived from the AGN luminosity". Does that mean the X-ray flux is not from real observations?

**Reply by Couleyr** [reply]
I see your point, meaning the X-ray data used here cannot be derived. It is disappointing that at such large column densities we cannot correct with these tools.

**Reply by Couleyr** [reply]
Yes, the X-ray data here is derived from AGN luminosity, we cannot give corrections because the column density is too large.

**Reply by Guang91** [reply]
If N_H<1e24, it's quite easy. But when N_H>1e24, Compton scattering makes things more complicated. You will need to model the X-ray spectrum first (but that would need quite high X-ray S/N) and additional work with XSPEC.

**Reply by Couleyr** [reply]
However, the signal-to-noise ratio of existing X-ray observations is insufficient.

**Reply by Guang91** [reply]
Then just drop the X-ray data. That's not useful.

**Reply by Couleyr** [reply]
Okay, thank you very much Mr. Yang.

**Reply by Couleyr** [reply]
According to your suggestion, we reconstructed the SED of the target source, and the result is shown in the figure. 
The question is why the corresponding residual of the X-ray point is not drawn in the lower panel of the figure.
[1266_best_model.pdf](https://github.com/mboquien/cigale/files/8464405/1266_best_model.pdf)

**Reply by Guang91** [reply]
It seems the X-ray points is near the x-axis limit. You can change the plotted range by, pcigale-plots sed --xrange 1e-4:6e2
